mathematica/retrobop_dynamical_stability.cpp

```diff
@@ -159,22 +158,15 @@ std::unique_ptr<Message> Read(std::ifstream& file) {
   return std::move(message);
 }
 
-HierarchicalSystem<Barycentric>::BarycentricSystem ReadSystem(
-    std::experimental::filesystem::path const& filename) {
-  std::ifstream input_file(filename, std::ios::in);
-  CHECK(!input_file.fail());
-  HierarchicalSystem<Barycentric>::BarycentricSystem system;
-  for (auto body = Read<serialization::MassiveBody>(input_file);
-       body != nullptr;
-       body = Read<serialization::MassiveBody>(input_file)) {
-    auto const degrees_of_freedom = Read<serialization::Pair>(input_file);
-    CHECK(degrees_of_freedom != nullptr);
-    system.bodies.push_back(MassiveBody::ReadFromMessage(*body));
-    system.degrees_of_freedom.push_back(
-        DegreesOfFreedom<Barycentric>::ReadFromMessage(*degrees_of_freedom));
-  }
-  input_file.close();
-  return system;
+HierarchicalSystem<Barycentric>::BarycentricSystem MakeStabilizedKSPSystem() {
+  static auto const& system = *[]() {
```

Comment:
The `&` worries me here.  Is the lifetime extended until the termination of the program?

Reply:
`new` so this lives forever, no lifetime extension going on here.
Since I'm using a lambda anyway, perhaps it would be clearer to `make_unique` and `release`.